src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -273,27 +275,40 @@ private PeerAddress Random(IEnumerable<PeerAddress> peers)
         public IEnumerable<PeerAddress> Attempted()
         {
             return this.peerAddresses.Values.Where(p =>
-                                p.Attempted &&
-                                p.ConnectionAttempts < PeerAddress.AttemptThreshold &&
-                                p.LastAttempt < this.dateTimeProvider.GetUtcNow().AddHours(-PeerAddress.AttempThresholdHours));
+                p.Attempted &&
+                p.ConnectionAttempts < PeerAddress.AttemptThreshold &&
+                p.LastAttempt < this.dateTimeProvider.GetUtcNow().AddHours(-PeerAddress.AttempThresholdHours) &&
+                !IsBanned(p));
         }
 
         /// <inheritdoc/>
         public IEnumerable<PeerAddress> Connected()
         {
-            return this.peerAddresses.Values.Where(p => p.Connected && p.LastConnectionSuccess < this.dateTimeProvider.GetUtcNow().AddSeconds(-60));
+            return this.peerAddresses.Values.Where(p => p.Connected && 
+                                                        p.LastConnectionSuccess < this.dateTimeProvider.GetUtcNow().AddSeconds(-60) &&
+                                                        !IsBanned(p));
         }
 
         /// <inheritdoc/>
         public IEnumerable<PeerAddress> Fresh()
         {
-            return this.peerAddresses.Values.Where(p => p.Fresh);
+            return this.peerAddresses.Values.Where(p => p.Fresh && !IsBanned(p));
         }
 
         /// <inheritdoc/>
         public IEnumerable<PeerAddress> Handshaked()
         {
-            return this.peerAddresses.Values.Where(p => p.Handshaked && p.LastConnectionHandshake < this.dateTimeProvider.GetUtcNow().AddSeconds(-60));
+            return this.peerAddresses.Values.Where(p => p.Handshaked && 
+                                                        p.LastConnectionHandshake < this.dateTimeProvider.GetUtcNow().AddSeconds(-60) &&
+                                                        !IsBanned(p));
+        }
+
+        /// <summary>
+        /// <c>True</c> if <see cref="PeerAddress.BanUntil"/> is in the future.
```

Comment:
`<c>true</c>`

but most importantly, the correct comment should say "if the address is banned" which is the real purpose of the method. what you say is how you implement it